src/Sparse/Row.jl

```diff
@@ -465,20 +467,23 @@ function scale_row!(a::SRow{T}, b::T) where T
       deleteat!(a.values, i)
```

Comment:
Orthogonal to this PR, but: Since you now handle `b==0` at the start, the `iszero(a.values[i])` check above the fold can only return `true` if the coefficient ring is not a domain. So it could be strengthened to something like `!is_domain_type(T) && iszero(a.values[i])`.

Since `is_domain_type` is a trait depending only on `T`, the compiler can eliminate the `is_domain_type(T)` check -- if it returns `true` it can elide the `if` block, and if it is `false` we the same code we have currently.

Reply:
But this can easily wait for a follow-up PR. (Also `scale_row!` and `scale_row_right!` could be merged)